File: legacy/legacy-apis/Psi.Models.ClientConfigurationModels/MobileConfigurations/Deals/Deals.cs
namespace Psi.Data.Models.ClientConfigurationModels.MobileConfigurations.Deals
{
    public class Deals : SettingsBaseHelper

    {
        private readonly ISettingsBase _settingsBase;
        private Cardlytics.Cardlytics _cardlytics;
        private RelevantSolutions.RelevantSolutions _relevantSolutions;
        private CheckingRewards.CheckingRewards _checkingRewards;

        public Deals(ISettingsBase settingsBase) : base(settingsBase)
        {
            _settingsBase = settingsBase;
        }

        public Cardlytics.Cardlytics Cardlytics
        {
            get { return _cardlytics ?? (_cardlytics = new Cardlytics.Cardlytics(_settingsBase)); }
            set { _cardlytics = value; }
        }

        public RelevantSolutions.RelevantSolutions RelevantSolutions
        {
            get { return _relevantSolutions ?? (_relevantSolutions = new RelevantSolutions.RelevantSolutions(_settingsBase)); }
            set { _relevantSolutions = value; }
        }

        public CheckingRewards.CheckingRewards CheckingRewards
        {
            get { return _checkingRewards ?? (_checkingRewards = new CheckingRewards.CheckingRewards(_settingsBase)); }
            set { _checkingRewards = value; }
        }
    }
}

File: infrastructure/models/ClientConfigurationModels/MobileConfigurations/Deals/Deals.ts
import { Setting, ISettingsGroup, ISettingsMetadata } from '@models/base/types';
import { Cardlytics } from '../Cardlytics.Cardlytics';
import { RelevantSolutions } from '../RelevantSolutions.RelevantSolutions';
import { CheckingRewards } from '../CheckingRewards.CheckingRewards';
export interface DealsConfig {
    Cardlytics: Cardlytics;
    RelevantSolutions: RelevantSolutions;
    CheckingRewards: CheckingRewards;
}

export class Deals implements ISettingsGroup {
    static readonly metadata: ISettingsMetadata = {
        groupName: 'Deals'
    };


            private _cardlytics: Cardlytics;
            get cardlytics(): Cardlytics {
                return this._cardlytics;
            }
            set cardlytics(value: Cardlytics) {
                this._cardlytics = value;
            }

            private _relevantSolutions: RelevantSolutions;
            get relevantSolutions(): RelevantSolutions {
                return this._relevantSolutions;
            }
            set relevantSolutions(value: RelevantSolutions) {
                this._relevantSolutions = value;
            }

            private _checkingRewards: CheckingRewards;
            get checkingRewards(): CheckingRewards {
                return this._checkingRewards;
            }
            set checkingRewards(value: CheckingRewards) {
                this._checkingRewards = value;
            }


    constructor() {}


        toSettings(): Setting[] {
            return [
                { key: "Deals.Cardlytics", value: this._cardlytics, dataType: 'cardlytics.cardlytics', label: "Cardlytics" },
                { key: "Deals.RelevantSolutions", value: this._relevantSolutions, dataType: 'relevantsolutions.relevantsolutions', label: "Relevant Solutions" },
                { key: "Deals.CheckingRewards", value: this._checkingRewards, dataType: 'checkingrewards.checkingrewards', label: "Checking Rewards" },
            ];
        }

}